ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -327,7 +334,9 @@ def _cumulative_to_window(translator, expr, window):
 
     klass = _cumulative_to_reduction[type(op)]
     new_op = klass(*op.args)
-    new_expr = expr._factory(new_op, name=expr._name)
+    new_expr = new_op.to_expr()
+    if expr.has_name():
+        new_expr = new_expr.name(expr.get_name())
```

Comment:
Another case of `has_name` that does have a test.

Reply:
All 4 `ibis/backends/tests/test_window.py::test_grouped_bounded_expanding_window` tests fail if I remove the `has_name()` check.